### PR TITLE
JAVA-2580: Remove 2.4 support

### PR DESCRIPTION
**Pull request by jyemin**
Nice to remove some of the cruft and shrink the Evergreen matrix.  Maybe this will make retryable writes a bit simpler in MixedBulkWriteOperation.

### CODE REVIEW

File: driver-async/src/examples/primer/UpdatePrimer.java [hunk 1]
```diff
@@ -23,15 +23,11 @@
 import com.mongodb.async.SingleResultCallback;
 import com.mongodb.client.result.UpdateResult;
 // @import: end
-import static com.mongodb.ClusterFixture.serverVersionAtLeast;
```
Comment (rozza) [inline driver-async/src/examples/primer/UpdatePrimer.java]: Do we even need these primer examples anymore?

File: driver-core/src/test/functional/com/mongodb/client/model/GeoFiltersFunctionalSpecification.groovy [hunk 2]
```diff
@@ -51,25 +49,21 @@ class GeoFiltersFunctionalSpecification extends OperationFunctionalSpecification
         find(nearSphere('geo', 1.01d, 1.01d, 0.1d, 0.0d)) == [firstPoint, thirdPoint]
     }
 
-    @IgnoreIf({ !serverVersionAtLeast(2, 4) })
```
Comment (rozza) [inline driver-core/src/test/functional/com/mongodb/client/model/GeoFiltersFunctionalSpecification.groovy]: :) wow!

File: driver-core/src/main/com/mongodb/operation/CreateIndexesOperation.java [hunk 3]
```diff
@@ -144,19 +138,12 @@ public Void execute(final WriteBinding binding) {
...
-                        executeWrappedCommandProtocol(binding, namespace.getDatabaseName(), getCommand(connection.getDescription()),
-                                connection, writeConcernErrorTransformer());
-                    } catch (MongoCommandException e) {
-                        throw checkForDuplicateKeyError(e);
-                    }
-                } else {
-                    if (requests.size() > 1) {
-                        throw new MongoInternalException("Creation of multiple indexes simultaneously not supported until MongoDB 2.6");
-                    }
-                    connection.insert(systemIndexes, true, WriteConcern.ACKNOWLEDGED, asList(new InsertRequest(getIndex(requests.get(0)))));
```
Comment (rozza) [inline driver-core/src/main/com/mongodb/operation/CreateIndexesOperation.java]: Can we deprecate these connection methods now?

File: driver-async/src/examples/primer/UpdatePrimer.java (hunk 1 above)
Comment (jyemin) [inline driver-async/src/examples/primer/UpdatePrimer.java]: Not sure, but no need to deal with that here.

File: driver-core/src/test/functional/com/mongodb/client/model/GeoFiltersFunctionalSpecification.groovy (hunk 2 above)
Comment (jyemin) [inline driver-core/src/test/functional/com/mongodb/client/model/GeoFiltersFunctionalSpecification.groovy]: Leftovers...

File: driver-core/src/main/com/mongodb/operation/CreateIndexesOperation.java (hunk 3 above)
Comment (jyemin) [inline driver-core/src/main/com/mongodb/operation/CreateIndexesOperation.java]: Let's do that once we start use OP_MSG for w:0 writes (JAVA-2585).

**Comment by jyemin** [comment]
Responded to all comments.

**Pull request closed without merging**